linkerd/app/inbound/src/direct.rs

```diff
@@ -137,20 +135,46 @@ impl<N> Inbound<N> {
                                 TransportHeader {
                                     port,
                                     name: None,
-                                    protocol: None,
+                                    protocol,
                                 } => {
                                     // When the transport header targets an alternate port (but does
                                     // not identify an alternate target name), we check the new
                                     // target's policy to determine whether the client can access
                                     // it.
                                     let addr = (client.local_addr.ip(), port).into();
-                                    let allow = policies.check_policy(OrigDstAddr(addr))?;
-                                    Ok(svc::Either::A(Local {
-                                        addr: Remote(ServerAddr(addr)),
-                                        policy: allow,
-                                        protocol: None,
-                                        client,
-                                    }))
+                                    let policy = policies.check_policy(OrigDstAddr(addr))?;
+                                    let local = match protocol {
+                                        None => {
+                                            let tls = tls::ConditionalServerTls::Some(
+                                                tls::ServerTls::Established {
+                                                    client_id: Some(client.client_id.clone()),
+                                                    negotiated_protocol: client.alpn,
+                                                },
+                                            );
+                                            let permit = policy
+                                                .check_authorized(client.client_addr, &tls)?;
+                                            svc::Either::A(LocalTcp {
+                                                addr: Remote(ServerAddr(addr)),
+                                                permit,
+                                                client_id: client.client_id,
+                                            })
+                                        }
+                                        Some(protocol) => {
+                                            // When TransportHeader includes the protocol, but does not
+                                            // include an alternate name we go through the Inbound HTTP
+                                            // stack.
+                                            let addr = (client.local_addr.ip(), port).into();
+                                            let policy =
+                                                policies.check_policy(OrigDstAddr(addr))?;
```

Comment:
Do we still need to check the policy here since we've done that already on L145? Can we re-use it here?

Reply:
correct, thanks. copy-paste error :/